pkg/apis/extensions/types.go

```diff
@@ -226,6 +226,10 @@ type DeploymentSpec struct {
 	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
 	// No label is added if this is set to empty string.
 	UniqueLabelKey string `json:"uniqueLabelKey,omitempty"`
+
+	// MaxProgressThresholdSeconds is the maximum time in seconds for a deployment
+	// to run before it is considered as failed.
+	MaxProgressThresholdSeconds *int `json:"maxProgressThresholdSeconds,omitempty"`
```

Comment:
As I commented [here](https://github.com/kubernetes/kubernetes/pull/19343/files#r49036843) we need to check our assumptions: I was under the impression that *any progress* should reset the timer and that we shouldn't treat this as a cumulative deployment time threshold.